packages/blade/src/components/Button/BaseButton/BaseButton.tsx

```diff
@@ -0,0 +1,259 @@
+import type { ReactElement } from 'react';
+import styled from 'styled-components';
+import getIn from '../../../utils/getIn';
+import type { BaseTextProps } from '../../Typography/BaseText';
+import BaseText from '../../Typography/BaseText';
+import type { Theme } from '../../BladeProvider';
+import { useTheme } from '../../BladeProvider';
+import type { IconComponent, IconProps, IconSize } from '../../Icons';
+import makeSpace from '../../../utils/makeSpace';
+import type { TypographyPlatforms } from '../../../tokens/global/typography';
+import makeBorderSize from '../../../utils/makeBorderSize';
+import StyledBaseButton from './StyledBaseButton';
+import type { ButtonMinHeight, ButtonTypography } from './buttonTokens';
+import { typography } from './buttonTokens';
+
+export type BaseButtonIcon = undefined | null | IconComponent;
+
+type BaseButtonCommonProps = {
+  variant?: 'primary' | 'secondary' | 'tertiary' | 'link';
+  intent?: 'positive' | 'negative' | 'notice' | 'info' | 'neutral';
+  contrast?: 'low' | 'high';
+  size?: 'large' | 'medium' | 'small' | 'xsmall';
+  iconPosition?: 'left' | 'right';
+  isDisabled?: boolean;
+  isFullWidth?: boolean;
+  onClick?: () => void;
+  type?: 'button' | 'reset' | 'submit';
+};
+
+type BaseButtonWithoutIconProps = BaseButtonCommonProps & {
+  icon?: undefined;
+  children: string;
+};
+
+type BaseButtonWithIconProps = BaseButtonCommonProps & {
+  icon: IconComponent;
+  children?: string;
+};
+
+export type BaseButtonProps = BaseButtonWithIconProps | BaseButtonWithoutIconProps;
+
+const StyledBaseText = styled(BaseText)(
+  ({
+    hasIcon,
+    iconPosition,
+    iconSpacing,
+  }: BaseTextProps &
+    Pick<BaseButtonProps, 'iconPosition'> & { hasIcon: boolean; iconSpacing: string }) => ({
+    paddingLeft: hasIcon && iconPosition === 'left' ? iconSpacing : '0px',
+    paddingRight: hasIcon && iconPosition === 'right' ? iconSpacing : '0px',
+  }),
+);
+
+type BaseButtonStyleProps = {
+  iconSize: IconSize;
+  fontSize: keyof Theme['typography']['fonts']['size'];
+  lineHeight: keyof Theme['typography']['lineHeights'];
+  minHeight: ButtonMinHeight;
+  iconSpacing: string;
+  iconColor: IconProps['color'];
+  textColor: BaseTextProps['color'];
+  spacing: string;
+  text?: string;
+  color: string;
```

Comment:
`color` is the button color. `textColor` is the text color for BaseText. We had it named as `buttonColor` before but changed it to `color` since it should be implicit that its for button unless explicitly mentioned like `textColor`

Reply:
didn't understand what do you mean by button color? ideally color and textColor are same thing, color in CSS world is color of the text. is there any possibility both of the these values will be different?

Reply:
Here color means the color of the button. I'm not mapping it exactly to the CSS property since it's the color of the button (`background-color`), hence `defaultColor`, `hoverColor`, `activeColor`, etc.